delivery_postlogistics/models/stock_picking.py

```diff
@@ -0,0 +1,458 @@
+
+from ..postlogistics.web_service import PostlogisticsWebService, sanitize_string
+
+_logger = logging.getLogger(__name__)
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    delivery_fixed_date = fields.Date(
+        "Fixed delivery date", help="Specific delivery date (ZAW3217)"
+    )
+
+    # TODO: consider refactoring these fields using a partner relation instead
+    delivery_place = fields.Char(help="For Deposit item service (ZAW3219)")
+    delivery_phone = fields.Char(
+        "Phone", help="For notify delivery by telephone (ZAW3213)"
+    )
+    delivery_mobile = fields.Char(
+        "Mobile", help="For notify delivery by telephone (ZAW3213)"
+    )
+
+    def _get_packages_from_picking(self):
+        # As CI doesn't allow warnings, we need to log as info here
+        # (deprecated decorator is unhelpfull).
+        _logger.info(
+            "This method will be removed in version 18.0. \
+Please use _get_quant_packages_from_picking instead."
+        )
+        # TODO: remove this method in version > 18.0
+        return self._get_quant_packages_from_picking()
+
+    def _get_quant_packages_from_picking(self):
+        """Get all the quant packages from the picking"""
+        self.ensure_one()
+        operations = self.env["stock.move.line"].search(
+            [
+                "|",
+                ("package_id", "!=", False),
+                ("result_package_id", "!=", False),
+                ("picking_id", "=", self.id),
+            ]
+        )
+        package_ids = set()
+        for operation in operations:
+            # Take the destination package. If empty, the package is
+            # moved so take the source one.
+            package_ids.add(operation.result_package_id.id or operation.package_id.id)
+
+        return self.env["stock.quant.package"].browse(package_ids)
+
+    def get_shipping_label_values(self, label):
+        # TODO: consider to depends on base_delivery_carrier_label
+        self.ensure_one()
+        return {
+            "name": label["name"],
+            "res_id": self.id,
+            "res_model": "stock.picking",
+            "datas": label["file"],
+            "file_type": label["file_type"],
+        }
+
+    def attach_shipping_label(self, label):
+        """Attach a label returned by generate_shipping_labels to a picking"""
+        if self.delivery_type != "postlogistics":
+            return super().attach_shipping_label(label)
+        self.ensure_one()
+        data = self.get_shipping_label_values(label)
+        context_attachment = self.env.context.copy()
+        # remove default_type setted for stock_picking
+        # as it would try to define default value of attachement
+        if "default_type" in context_attachment:
+            del context_attachment["default_type"]
+        return (
+            self.env["shipping.label"].with_context(**context_attachment).create(data)
+        )
+
+    def postlogistics_cod_amount(self):
```

Comment:
here the `_set_a_default_package` is missing
https://github.com/OCA/delivery-carrier/blob/5a4f7c4eb5f7bd2c6bd0298d69937f0199270c02/delivery_postlogistics/models/stock_picking.py#L78

Reply:
I see it's removed now, so nevermind